### PR TITLE
83 add encryption key to create identity

### PR DESCRIPTION
**Pull request by Schereo**
Shared-modules
- Added `encrypt` field to `IdentityKeys` with X25519 keys for encryption
- Published new version and updated the changelog

SSI-bridge
- Added encryption keys to identity
- Removed Account API

### CODE REVIEW

File: api/shared-modules/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+### 0.2.2 (2022-08-05)
+
+- Made `encrypt` field in `IdentityKeys` optional
+
+### 0.2.1 (2022-08-05)
+
+- Added `encrypt` field to `IdentityKeys` with X25519 keys for encryption
```
Comment (dominic22) [inline api/shared-modules/CHANGELOG.md:9]: please use alpha version instead of 0.2.1 and 0.2.2

File: api/ssi-bridge/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+### 0.1.9 (2022-08-05)
```
Comment (dominic22) [inline api/ssi-bridge/CHANGELOG.md:3]: please remove this I will add a changelog here with the increase to 0.2.0

File: api/ssi-bridge/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "ssi-bridge",
-  "version": "0.1.8",
+  "version": "0.1.9",
```
Comment (dominic22) [inline api/ssi-bridge/package.json:3]: please remove this I increase the version when the identity update is completed here with the increase to 0.2.0

File: api/ssi-bridge/src/services/ssi-service.ts [hunk 4]
```diff
@@ -203,31 +213,40 @@ export class SsiService {
...
+				document.id(),
+				encryptionKeyPair.type(),
+				encryptionKeyPair.public(),
+				'kex-0'
+			);
+			document.insertMethod(encryptionMethod, MethodScope.KeyAgreement());
+
+			document.signSelf(signKeyPair, document.defaultSigningMethod().id());
+			const client = await this.getIdentityClient(false);
+			await client.publishDocument(document);
```
Comment (dominic22) [inline api/ssi-bridge/src/services/ssi-service.ts:237]: please use method publishSignedDoc

File: api/ssi-bridge/src/services/ssi-service.ts [hunk 5]
```diff
@@ -203,31 +213,40 @@ export class SsiService {
...
+			const encryptionMethod = new Identity.VerificationMethod(
+				document.id(),
+				encryptionKeyPair.type(),
+				encryptionKeyPair.public(),
+				'kex-0'
+			);
+			document.insertMethod(encryptionMethod, MethodScope.KeyAgreement());
+
+			document.signSelf(signKeyPair, document.defaultSigningMethod().id());
+			const client = await this.getIdentityClient(false);
```
Comment (dominic22) [inline api/ssi-bridge/src/services/ssi-service.ts:236]: please move this to beginning of the function

File: api/ssi-bridge/src/services/ssi-service.ts [hunk 6]
```diff
@@ -203,31 +213,40 @@ export class SsiService {
...
+			const encryptionKeyPair = new KeyPair(KeyType.X25519);
+			const encryptionMethod = new Identity.VerificationMethod(
+				document.id(),
+				encryptionKeyPair.type(),
+				encryptionKeyPair.public(),
+				'kex-0'
+			);
+			document.insertMethod(encryptionMethod, MethodScope.KeyAgreement());
+
+			document.signSelf(signKeyPair, document.defaultSigningMethod().id());
```
Comment (dominic22) [inline api/ssi-bridge/src/services/ssi-service.ts:235]: cant we use publishSignedDoc here?

Comment (dominic22) [reply api/ssi-bridge/src/services/ssi-service.ts:235]: I think you need to pass prevMessageId as undefined and add a check into the publishSignedDoc and only set:

`if(prevMessageId) { newDoc.setMetadataPreviousMessageId(prevMessageId);}`

File: api/ssi-bridge/src/services/ssi-service.ts [hunk 7]
```diff
@@ -203,31 +213,40 @@ export class SsiService {
...
+			const client = await this.getIdentityClient(false);
+			await client.publishDocument(document);
 
 			return {
-				account,
-				doc: account.document(),
-				key: keyPair
+				doc: document,
+				key: signKeyPair,
```
Comment (dominic22) [inline api/ssi-bridge/src/services/ssi-service.ts:241]: can you please rename `key` to `signingKeys` or `signatureKeys` and encryption to `encryptionKeys` ?

**Review by JuriBog — APPROVED** [review]

File: api/ssi-bridge/src/services/ssi-service.ts [hunk 8]
```diff
@@ -203,31 +213,37 @@ export class SsiService {
...
+			const signingKeyPair = new KeyPair(KeyType.Ed25519);
+			const document = new Document(signingKeyPair, this.getConfig(false).network.name());
+
+			// Add encryption keys and capabilities to Identity
+			const encryptionKeyPair = new KeyPair(KeyType.X25519);
+			const encryptionMethod = new Identity.VerificationMethod(
+				document.id(),
+				encryptionKeyPair.type(),
+				encryptionKeyPair.public(),
+				'kex-0'
```
Comment (dominic22) [inline api/ssi-bridge/src/services/ssi-service.ts:231]: could you please make this for now a const var defined at top of the method... we might make this later an enum

**Review by dominic22 — APPROVED** [review]

**Pull request merged**